### ISSUE TITLE
Add a docs for HMR integration for a development

### ISSUE THREAD

**Issue by vecernik** [datorama/akita#108]
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[ ] Feature request
[X] Documentation issue or request
[X] Support request
[ ] Other... Please describe:
</code></pre>

## Current behavior
How to run Akita/Angular-cli project with HotModuleReplacement technique?

## Expected behavior
Get to the same state after auto-refresh when a code changes during a development, possibly using a snapshot manager.

## What is the motivation / use case for changing the behavior?
Drammatically reduce a time between page reloads, when i.e. only css changes.

## Environment

<pre><code>
Angular version: 7

Browser: all
 
For Tooling issues:
- Node version: 10
- Platform:  any
</code></pre>

**Comment by NetanelBasal** [comment]
Did you work with this https://github.com/angular/angular-cli/wiki/stories-configure-hmr?

**Comment by iwnow** [comment]
@NetanelBasal yes, without using this plugin, everything works fine

**Comment by NetanelBasal** [comment]
I'll try to investigate, thanks.

**Comment by NetanelBasal** [comment]
@iwnow can you try to destroy the plugin? 

```ts
export const hmrBootstrap = (module: any, bootstrap: () => Promise<NgModuleRef<any>>) => {
  let ngModule: NgModuleRef<any>;
  module.hot.accept();
  bootstrap().then(mod => ngModule = mod);

  module.hot.dispose(() => {
    const appRef: ApplicationRef = ngModule.injector.get(ApplicationRef);
    const elements = appRef.components.map(c => c.location.nativeElement);
    const makeVisible = createNewHosts(elements);
    ngModule.destroy();
    makeVisible();
   presistState.destroy(); <======
  });
};
```

**Comment by iwnow** [comment]
@NetanelBasal i tried but still error received
and after hmr reload cycle all data getting default, because localStorage disconnect after destroy
i think after destroy we need recreate plugin as well
or don't ever call the destroy method on plugin and after ngModule recreate then akita Store need reapply the current instance plugin itself

**Comment by NetanelBasal** [comment]
@iwnow I set up a clean project with Akita, [hmr](https://github.com/angular/angular-cli/wiki/stories-configure-hmr), and persistState and it works fine. I can't see the error you're getting. Anything that I could do to reproduce it?

**Comment by iwnow** [comment]
> @iwnow I set up a clean project with Akita, [hmr](https://github.com/angular/angular-cli/wiki/stories-configure-hmr), and persistState and it works fine. I can't see the error you're getting. Anything that I can do to reproduce it?

@NetanelBasal what version you used?

**Comment by NetanelBasal** [comment]
Akita latest version and Angular v7.

**Comment by TheAifam5** [comment]
Also I have the issue with HMR witout persist state enabled:
```
TypeError: "__stores__[action.payload.name] is undefined; can't access its "_value" property"
```

**Comment by NetanelBasal** [comment]
@TheAifam5  you are using Akita latest version and Angular v7?

**Comment by TheAifam5** [comment]
@NetanelBasal yes, I'm using the latest version which is `1.19.0` and akita-ng-router-store `1.0.1`.

**Comment by NetanelBasal** [comment]
As I already wrote, I created a new project with these tools, and I didn't get any error. It will be helpful if you can reproduce it with a live example.

**Comment by TheAifam5** [comment]
@NetanelBasal yup, i gonna create one for you.

**Comment by NetanelBasal** [comment]
Thanks, appreciate.

**Comment by TheAifam5** [comment]
@NetanelBasal I wasn't able to make it in `stackblitz.com` because doesn't support a proper HMR from webpack-dev-server.

Here is reproduction of local-hosted project with the issue which I described above.

[store-hmr-108-repro.zip](https://github.com/datorama/akita/files/2576083/store-hmr-108-repro.zip)

**Comment by NetanelBasal** [comment]
Thanks, I will check it soon.

**Comment by NetanelBasal** [comment]
I see the problem. When we destroy the module, it calls `ngOnDestroy` on the providers, and Akita deletes the store. 

I guess we need some indication that we are in `hmr` mode, something like:
```ts
export const hmrBootstrap = (
  module: any,
  bootstrap: () => Promise<NgModuleRef<any>>
) => {
  let ngModule: NgModuleRef<any>;
  module.hot.accept();
  bootstrap().then(mod => (ngModule = mod));
  window.hmr = true; <=====
  module.hot.dispose(() => {
    const appRef: ApplicationRef = ngModule.injector.get(ApplicationRef);
    const elements = appRef.components.map(c => c.location.nativeElement);
    const makeVisible = createNewHosts(elements);
    ngModule.destroy();
    makeVisible();
    window.hmr = false; <=====
  });
};
```

**Comment by TheAifam5** [comment]
@NetanelBasal actually HMR mode uses `module['hmr']`.

**Comment by NetanelBasal** [comment]
Yes, I know, but I can't count on this, as I want to call ngOnDestroy when it comes from the regular flow and not triggered by hmr.

**Comment by NetanelBasal** [comment]
https://netbasal.gitbook.io/akita/angular-plugins/hmr
